R/cut_dataToWaterYear.R

```diff
@@ -0,0 +1,48 @@
+#' Cuts the discharge time series to full water years
+#' @param x A dataframe containing a vector of date values in the first 
+#' column and vector of numeric flow values in the second column.
+#' @param wyMonth The month of the year in which the water year starts. The water year begins on the first day of wyMonth.
+#' @return data.frame in original structure, but cut to full water years
+#' @details To ensure the input data is valid for use with other 
+#' EflowStats functions. 
+#' #' \enumerate{
+#'   \item First column must be of class `Date`.
+#'   \item Second must be of class `numeric`.
+#'   \item `wyMonth`input must be of class `integer`. 
+#' }
+#' @examples
+#' x <- sampleData[, c('date', 'discharge')]
+#' cut_dataToWaterYear(x,10L)
+#' @export
+
+
+
+cut_dataToWaterYear <- function(x,wyMonth=10L) {
+  ###rename dataframe for convenient use inside function
+  names(x) <- c("date","discharge")
+  
+  ###Order by date
+  x = dplyr::arrange(x, date)
+  x$year_val <- get_waterYear(x[,1], wyMonth)
+  
+  # get first and last water year integer of time series
+  first_year <- x$year_val[1]
+  last_year <- x$year_val[length(x[,1])]
+  
+  # get the current days of the first and last water year of time series
+  ndays_first_year <- length(x[x$year_val == first_year,][,1])
+  ndays_last_year <- length(x[x$year_val == last_year,][,1])
+  
+  # get the target number of days (depends if water year is in a leap year or not)
+  ndays_first_year_target <- ifelse(((first_year %% 4 == 0) & (first_year %% 100 != 0)) | (first_year %% 400 == 0), 366, 365)
+  ndays_last_year_target <- ifelse(((last_year %% 4 == 0) & (last_year %% 100 != 0)) | (last_year %% 400 == 0), 366, 365)
```

Comment:
I'm not following this leap year check. Can you add comments to describe? Why can't we use only (first_year %% 4 == 0)? Or lubridate's leap_year function?

Could also make this a function because it's applied at least these 2 times.